lib/rules/declaration-property-value-keyword-no-deprecated/index.mjs

```diff
@@ -0,0 +1,174 @@
+	fixable: true,
+};
+
+const DEPRECATED_COLORS = {
+	activecaption: 'canvas',
+	appworkspace: 'canvas',
+	background: 'canvas',
+	inactivecaption: 'canvas',
+	infobackground: 'canvas',
+	menu: 'canvas',
+	scrollbar: 'canvas',
+	window: 'canvas',
+	activeborder: 'ButtonBorder',
+	inactiveborder: 'ButtonBorder',
+	threeddarkshadow: 'ButtonBorder',
+	threedhighlight: 'ButtonBorder',
+	threedlightshadow: 'ButtonBorder',
+	threedshadow: 'ButtonBorder',
+	windowframe: 'ButtonBorder',
+	captiontext: 'CanvasText',
+	infotext: 'CanvasText',
+	menutext: 'CanvasText',
+	windowtext: 'CanvasText',
+	buttonhighlight: 'ButtonFace',
+	buttonshadow: 'ButtonFace',
+	threedface: 'ButtonFace',
+	inactivecaptiontext: 'GrayText',
+};
+const COLOR_PROPERTIES = [...singleValueColorProperties.values()].map(
+	(value) => /** @type {[string, Record<string, string>]} */ ([value, DEPRECATED_COLORS]),
+);
+const PROPERTY_NAME_TO_KEYWORDS = new Map([
+	[
+		'appearance',
+		/** @type {string[] | Record<string, string>} */ ({
+			button: 'auto',
+			checkbox: 'auto',
+			listbox: 'auto',
+			menulist: 'auto',
+			meter: 'auto',
+			'progress-bar': 'auto',
+			'push-button': 'auto',
+			radio: 'auto',
+			searchfield: 'auto',
+			'slider-horizontal': 'auto',
+			'square-button': 'auto',
+			textarea: 'auto',
+		}),
+	],
+	[
+		'image-rendering',
+		{
+			optimizeQuality: 'smooth',
+			optimizeSpeed: 'pixelated',
+		},
+	],
+	['overflow', { overlay: 'auto' }],
+	['text-justify', { distribute: 'inter-character' }],
+	['text-orientation', { 'sideways-right': 'sideways' }],
+	['user-select', { element: 'contain' }],
+	['zoom', { reset: '1' }],
+	['text-decoration-line', ['blink']],
+	['box-sizing', ['padding-box']],
+	['image-orientation', ['flip']],
+	['min-height', ['intrinsic', 'min-intrinsic']],
+	['min-width', ['intrinsic', 'min-intrinsic']],
+	['max-height', ['intrinsic', 'min-intrinsic']],
+	['max-width', ['intrinsic', 'min-intrinsic']],
+	['height', ['intrinsic', 'min-intrinsic']],
+	['width', ['intrinsic', 'min-intrinsic']],
+	['word-break', ['break-word']],
+	...COLOR_PROPERTIES,
+]);
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary, secondaryOptions) => {
+	return (root, result) => {
+		const validOptions = validateOptions(
+			result,
+			ruleName,
+			{ actual: primary },
+			{
+				optional: true,
+				actual: secondaryOptions,
+				possible: {
+					ignoreKeywords: [isString, isRegExp],
+				},
+			},
+		);
+
+		if (!validOptions) return;
+
+		root.walkDecls((decl) => {
+			if (!isStandardSyntaxDeclaration(decl)) return;
+
+			const { prop, value } = decl;
+
+			if (optionsMatches(secondaryOptions, 'ignoreKeywords', value)) return;
+
+			const keywords = PROPERTY_NAME_TO_KEYWORDS.get(prop.toLowerCase());
+
+			if (!keywords) return;
+
+			const parsedValue = valueParser(getDeclarationValue(decl));
+
+			parsedValue.walk((node) => {
+				if (node.type !== 'word') return;
+
+				const actualKeyword = node.value.toLowerCase();
+
```

Comment:
```js
let message;
let messageArgs;
let fix;

if (keywordMaps) {
	const expectedKeyword = keywordMaps[actualKeyword];

	if (!expectedKeyword) return;

	// Setting message, messageArgs, and fix...
} else if (keywordLists)
	if (!keywordLists.includes(actualKeyword)) return;

	// Setting message and messageArgs...
} else {
	return;
}
```

Reply:
Ill partially apply this because the `else if` permits to skip the `isPlainObject(keywords)` evaluation.